servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -2664,6 +2664,36 @@ public final Publisher<T> skipWhile(Predicate<? super T> predicate) {
         return filter(FilterPublisher.skipWhileSupplier(predicate));
     }
 
+    /**
+     * After the first element of this {@link Publisher} is emitted pass it, and a {@link Publisher} representing the
+     * remainder of the stream, to a function to a mapping function.
+     * <p>
+     * <pre>{@code
+     *     class Result {
+     *         Result(T head, Iterator<T> tail) {
+     *             this.head = head;
+     *             this.tail = tail;
+     *         }
+     *
+     *         T head;
+     *         Iterator<T> tail;
+     *     }
+     *
+     *     Iterator<T> itr = resultOfThisPublisher();
+     *     Result result = new Result(itr.next(), itr);
+     *     return result;
+     * }</pre>
+     *
+     * @param packer A function that takes the head of the input stream and processes it, along with a {@link Publisher}
+     * of the remainder of the stream.
```

Comment:
We should document a note that the `packer` itself and any following operator that works with `R` absolutely must take care of the tail publisher if they decide to throw an exception. Users should either drain it or subscribe and cancel. Otherwise, they may leak resources.